### PR TITLE
add: Relationship between Event component and other k8s components

### PR DESCRIPTION
**Pull request by RipulHandoo**
**Notes for Reviewers**

This PR:
1. Add Relationship between k8s Event resource and other k8s resources.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by RipulHandoo** [comment]
@MUzairS15 In the video, I've added all the relationships for which I raised a PR. However, I manually added the values for now.
https://github.com/meshery/meshery/assets/107461226/70cae644-a4bf-4daa-9b58-8dc3c4c1b705

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -254,6 +254,126 @@
...
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["name"],
+                ["namespace"],
+                ["kind"],
+                ["apiVersion"]
+              ],
+              "description": "A Pod "
```
Comment (RipulHandoo) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:272]: Have to add description now.

**Review by MUzairS15 — COMMENTED** [review]
@RipulHandoo Sync your branch with master, now you no longer be required to manually enter values .
Thanks for the recording, it is helpful.
But will you please update the video on all PRs in which you create relationship as they should be created?

**Review by MUzairS15 — COMMENTED** [review]
@RipulHandoo, In the recording please show,
1. How are you adding values?
2. Config update after relationships have been created.
3. Trying fiddling with existing relationships.
4. Change the config of the old relationships such that the components get un-related.
Eg: `For pod->pvc->pv change the label of pv, or selector labels of PVC or volumeName in pod/pvc ....`, similarly `For role/cr->rolebinding/crb->serviceAccount try changing the serviceAccountname or roleRef`
5. Again update the config of the old relationships to what it was, and ensure the relationship gets auto-created.

**Comment by MUzairS15** [comment]
@RipulHandoo Is spreadsheet updated?

**Comment by RipulHandoo** [comment]
> @RipulHandoo Sync your branch with master, now you no longer be required to manually enter values . Thanks for the recording, it is helpful. But will you please update the video on all PRs in which you create relationship as they should be created?

@MUzairS15 I've synced my branch with the master. However, when I bring a pod near a configMap or vice versa, the configuration is not getting updated. The same issue persists for the relations I raised PRs for.
Here is how i'm implementing it.

https://github.com/meshery/meshery/assets/107461226/5d558e77-4efc-4cd1-8b4f-b9839d993954

**Review by leecalcote — APPROVED** [review]

**Pull request merged**